Spigot-API-Patches/0167-Add-BaseComponent-array-to-a-variety-of-events.patch

```diff
@@ -0,0 +1,2196 @@
++ * from 1, and with 1 referring to the first argument. The $ suffix is mandatory to describe this parameter.
++ * </p>
++ *
++ * <p>
++ * The flags parameter is a set of characters that will modify the output format. The valid flags
++ * are as follows:
++ * <ul>
++ * <li>
++ *     '-' (left justified flag)
++ * </li>
++ * <li>
++ *     '<' (reuse argument flag)
++ * </li>
++ * <li>
++ *     'E' (passon everything flag)
++ * </li>
++ * <li>
++ *     'F' (passon text formatting flag)
++ * </li>
++ * <li>
++ *     'I' (passon nothing flag)
++ * </li>
++ * <li>
++ *     'e' (retain everything flag)
++ * </li>
++ * <li>
++ *     'f' (retain text formatting flag)
++ * </li>
++ * <li>
++ *     'i' (retain nothing flag)
++ * </li>
++ * </ul>
++ * </p>
++ *
++ * <p>
++ * The flag '-' specifies that the result will be left-justified. If this is not specified,
++ * the result is right-justified.
++ * </p>
++ *
++ * <p>
++ * The flag '<' specifies that the previous format specifier's argument index be used. Note that
++ * specifying an argument_index when this flag is present is considered invalid. This flag may not be used
++ * in the first format specifier.
++ * </p>
++ *
++ * <p>
++ * There can be no more than 1 passon flag present and no more than 1 retain flag present.
++ * </p>
++ *
++ * <p>
++ * If a passon flag is not specified, then 'F' is used instead.
++ * </p>
++ *
++ * <p>
++ * If a retain flag is not specified, then 'f' is used instead.
++ * </p>
++ *
++ * <p>
++ * Passon flags refer to the type of formatting to pass on to the strings on the right of the specifier.
++ * </p>
++ *
++ * <p>
++ * Retain flags refer to the type of formatting to retain from the formatting to the left of the specifier.
++ * </p>
++ *
++ * <p>
++ * The width parameter specifies the minimum number of characters to be written. In this case, the padding
++ * is just spaces. The default value for width is 0.
++ * </p>
++ *
++ * <p>
++ * The precision flag specifies the maximum number of characters to be written in this case. The value is trimmed
++ * according to {@link BaseComponentUtils#trimMessage(BaseComponent[], int)}.
++ * The default value for precision is {@link Integer#MAX_VALUE}.
++ * </p>
++ *
++ * <p>
++ * The conversion parameter is only one of the following:
++ * <ul>
++ * <li>
++ *     's'
++ * </li>
++ * <li>
++ *     'S'
++ * </li>
++ * <li>
++ *     '%'
++ * </li>
++ * <li>
++ *     'n'
++ * </li>
++ * </ul>
++ * </p>
++ *
++ * <p>
++ * If the conversion is 'n' or '%', then specifying the precision parameter or argument index is invalid. If the
++ * width or flags argument is present, they are simply ignored.
++ * </p>
++ *
++ * <p>
++ * For conversion 's' or 'S', a BaseComponent[] argument is required. If the 'S' conversion is used, then the
++ * BaseComponent[] argument is converted to uppercase via {@link BaseComponentUtils#toUpperCase(BaseComponent[])}.
++ * </p>
++ *
++ * <p>
```

Comment:
explicitly state the order at which the format is iterated (currently it should be iterative)

Reply:
nevermind make it undefined